packages/smooth_app/lib/pages/all_product_list_page.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_list_tile.dart';
+import 'package:smooth_app/pages/product/common/product_query_page_helper.dart';
+import 'package:smooth_app/pages/product_list_user_dialog_helper.dart';
+import 'package:smooth_app/widgets/smooth_app_bar.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+
+/// Page that lists all product lists.
+class AllProductListPage extends StatefulWidget {
+  const AllProductListPage();
+
+  @override
+  State<AllProductListPage> createState() => _AllProductListPageState();
+}
+
+class _AllProductListPageState extends State<AllProductListPage> {
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
+    final DaoProductList daoProductList = DaoProductList(localDatabase);
+    final List<ProductList> productLists = <ProductList>[
+      ProductList.scanSession(),
+      ProductList.scanHistory(),
+      ProductList.history(),
+    ];
+    final List<String> userLists = daoProductList.getUserLists();
+    for (final String userList in userLists) {
+      productLists.add(ProductList.user(userList));
+    }
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    return SmoothScaffold(
+      appBar: SmoothAppBar(title: Text('Select a list')),
+      body: ListView.builder(
```

Comment:
Generally speaking, we don't use them, but we should always provide an `itemExtent` an `ListView.builder`